libs/matchers/hostname.js

```diff
@@ -1,5 +1,6 @@
 var match = require('./util').match;
 
 module.exports = function(req, value) {
-  return match('hostname', value, req.headers.host.split(':')[0], true);
+  var host = req.headers.host || req.headers[':authority'];
```

Comment:
technically `user@` can be supplied to authority, but we don't support that behavior (yet).

Reply:
At some point it might be nice to have a helper, ala:

```
const { user, host, port } = util.getAuthority(req.headers[':authority']);
```

Reply:
I'm seeing conflicting info on that in the [HTTP2 RFC](https://http2.github.io/http2-spec/#rfc.section.8.1.2.3):

> The :authority pseudo-header field includes the authority portion of the target URI ([RFC3986], Section 3.2). The authority MUST NOT include the deprecated userinfo subcomponent for http or https schemed URIs.

Reply:
I will extract the host/port parsing into a util, though.